figures/scripts/supp-S3-panels-BCD.R

```diff
@@ -0,0 +1,321 @@
+# Stephanie J. Spielman for CCDL, 2022
+
+# This script creates three S3 panels (B, C, D)
+
+
+## Load libraries ------------------------
+library(tidyverse)
+library(ComplexHeatmap)
+
+
+## Define paths  -----------------------
+root_dir <- rprojroot::find_root(rprojroot::has_dir(".git"))
+data_dir <- file.path(root_dir, "data")
+analyses_dir <- file.path(root_dir, "analyses")
+scratch_dir <- file.path(root_dir, "scratch")
+figure_dir <- file.path(root_dir, "figures")
+palettes_dir <- file.path(figure_dir, "palettes")
+supp_figures_dir <- file.path(figure_dir, "pdfs", "supp")
+
+## Define output files
+figure_s3b_file <- file.path(supp_figures_dir, "supp_figS3-B.pdf")
+figure_s3c_file <- file.path(supp_figures_dir, "supp_figS3-C.pdf")
+figure_s3d_file <- file.path(supp_figures_dir, "supp_figS3-D.pdf")
+
+## Figure S3B ----------------------------
+# Adapted from https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/master/analyses/cnv-chrom-plot/cn_status_heatmap.Rmd
+
+# Define paths for this figure
+original_dir <- file.path(analyses_dir, "cnv-chrom-plot")
+cnv_dir <- file.path(analyses_dir, "copy_number_consensus_call", "results")
+
+# Import bin calls
+bin_calls_df <- read_tsv(file.path(original_dir, "results", "cn_status_bp_per_bin.tsv"))
+
+# Import color palettes
+histology_label_mapping <- readr::read_tsv(file.path(palettes_dir, "broad_histology_cancer_group_palette.tsv"))
+
+# array of the three hex codes
+divergent_col_hex <- readr::read_tsv(
+  file.path(figure_dir, "palettes", "divergent_color_palette.tsv")
+  ) %>%
+  # Only keep only these three colors - we only need three values for `gain`, `neutral`, and `loss`.
+  filter(
+    color_names %in% c("divergent_low_4", "divergent_neutral", "divergent_high_4")
+  ) %>%
+  pull(hex_codes)
+
+
+# Read in metadata and merge with colors
+metadata <- read_tsv(file.path(data_dir, "pbta-histologies.tsv"), guess_max = 10000) %>%
+  # Join on the colors 
+  left_join(histology_label_mapping, by = "cancer_group") %>%
+  # Select what is needed
+  select(contains("cancer_group"), 
+                Kids_First_Biospecimen_ID, 
+                tumor_ploidy) %>%
+  # remove "Other"
+  filter(cancer_group != "Other")
+
+
+### Set up chromosomal sizes for making bins. 
+chr_sizes <- read_tsv(file.path(data_dir, "WGS.hg38.strelka2.unpadded.bed"),
+                             col_names = c("chrom", "start", "end")
+) %>%
+  # Reformat the chromosome variable to drop the "chr"
+  mutate(chrom = factor(gsub("chr", "", chrom),
+                               levels = c(1:22, "X", "Y", "M")
+  )) %>%
+  # Remove sex chromosomes
+  filter(!(chrom %in% c("X", "Y", "M")))
```

Comment:
This all looks consistent with https://github.com/sjspielman/OpenPBTA-analysis/blob/3655669f6363494a6281288919c4db20a65b1aa2/analyses/cnv-chrom-plot/cn_status_heatmap.Rmd#L205-L213